agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AzureMonitorLogProcessor.java

```diff
@@ -31,4 +41,65 @@ public void onEmit(Context context, ReadWriteLogRecord logRecord) {
       logRecord.setAttribute(AiSemanticAttributes.ITEM_COUNT, itemCount);
     }
   }
+
+  @Nullable
+  private static Class<?> getSdkReadWriteLogRecordClass() {
+    try {
+      return Class.forName("io.opentelemetry.sdk.logs.SdkReadWriteLogRecord");
+    } catch (ClassNotFoundException e) {
+      return null;
+    }
+  }
+
+  @Nullable
+  private static Field getLockField() {
+    if (sdkReadWriteLogRecordClass == null) {
+      return null;
+    }
+    try {
+      Field lockField = sdkReadWriteLogRecordClass.getDeclaredField("lock");
+      lockField.setAccessible(true);
+      return lockField;
+    } catch (NoSuchFieldException e) {
+      return null;
+    }
+  }
+
+  @Nullable
+  private static Field getAttributesMapField() {
+    if (sdkReadWriteLogRecordClass == null) {
+      return null;
+    }
+    try {
+      Field attributesMapField = sdkReadWriteLogRecordClass.getDeclaredField("attributes");
+      attributesMapField.setAccessible(true);
+      return attributesMapField;
+    } catch (NoSuchFieldException e) {
+      return null;
+    }
+  }
+
+  private static void setAttributeExceptionLogged(Span span, ReadWriteLogRecord logRecord) {
+    if (lockField != null && attributesMapField != null) {
```

Comment:
(small suggestion, early return will avoid extra indentation)
```suggestion
    if (lockField == null || attributesMapField == null) {
      return;
    }
```

Reply:
not sure what you mean, my suggestion is just a refactoring to avoid extra indentation, doesn't change logic